### PR TITLE
Complete args for all jobs

### PR DESCRIPTION
**Pull request by hongkailiu**
/cc @openshift/openshift-team-developer-productivity-test-platform 

/hold

wait for the change on release repo

---
Update:
https://github.com/openshift/release/pull/7107

### CODE REVIEW

File: test/prowgen-integration/data/output/jobs/private/duper/private-duper-master-periodics.yaml [hunk 1]
```diff
@@ -19,6 +19,8 @@ periodics:
       - --artifact-dir=$(ARTIFACTS)
       - --branch=master
       - --give-pr-author-access-to-namespace=true
+      - --image-import-pull-secret=/etc/pull-secret/.dockerconfigjson
+      - --kubeconfig=/etc/apici/kubeconfig
```
[the remark this answers is not in the document]
Comment (hongkailiu) [reply test/prowgen-integration/data/output/jobs/private/duper/private-duper-master-periodics.yaml]: All of these are ramifications of build farm, i.e., the cluster of running prow pods and the one of running the test pods are different.
- `kubeconfig`: for imagestream client to access the image in another cluster
- `image-import...`: for build pod to access the image in another cluster

File: cmd/ci-operator-prowgen/main.go [hunk 2]
```diff
@@ -59,6 +58,8 @@ exec ci-operator \
 	oauthTokenPath  = "/usr/local/github-credentials"
 	oauthSecretName = "github-credentials-openshift-ci-robot-private-git-cloner"
 	oauthKey        = "oauth"
+
+	build01Context = "ci/api-build01-ci-devcluster-openshift-com:6443"
```
[the remark this answers is not in the document]
Comment (hongkailiu) [reply cmd/ci-operator-prowgen/main.go]: Discussed over slack.

File: cmd/ci-operator-prowgen/main_test.go [hunk 3]
```diff
@@ -55,6 +55,8 @@ func TestGeneratePodSpec(t *testing.T) {
 						"--org=org",
 						"--repo=repo",
 						"--branch=branch",
+						"--kubeconfig=/etc/apici/kubeconfig",
```
[the remark this answers is not in the document]
Comment (hongkailiu) [reply cmd/ci-operator-prowgen/main_test.go]: Nice article.

I have to admit that I spent almost a whole day to fix the unit/integration tests, reading the diff output by `t.errorf(...)`.

For the integration tests, you kind of use `golden files` already, and the missing part is the `update` flag. My trick is to output files to overwrite the expected files. That is my manual implementation of that flag. Not efficient, I realized too.

**Comment by stevekuznetsov** [comment]
LGTM once we pass tests

**Pull request merged**